### PR TITLE
TST: Jenkins bye bye bye

### LINKED ISSUE #50: TST: ssbjenkins is no more, Jenkinsfile now useless
FYI

### PR DESCRIPTION
**Pull request by pllim**
Fix #50 

- [x] Make sure it passes at https://github.com/pllim/ci_watson/actions -- See https://github.com/pllim/ci_watson/runs/2964215119?check_suite_focus=true

Actions will not run in this repo until after merge.

Pro: No longer relies on ~Jenkins~ ssbjenkins.
Con: Functionality related to Artifactory is now untested in push/PR events; i.e., `--big-data`.

Out of scope:

* Open up Artifactory access to Actions.
* Use `tox`.

### CODE REVIEW

File: .github/workflows/ci_workflows.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+on:
+  push:
+  pull_request:
+  schedule:
+    # Weekly Wednesday 7 AM build
+    # * is a special character in YAML so you have to quote this string
+    - cron: '0 7 * * 3'
+
+env:
+  TEST_BIGDATA: "https://bytesalad.stsci.edu/artifactory"
```
Comment (jdavies-st) [inline .github/workflows/ci_workflows.yml:12]: Since the workflow cannot access our Artifactory server, probably no reason to have this defined.

File: .github/workflows/ci_workflows.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    - name: Lint with flake8
+      run: |
+        python -m pip install --upgrade pip flake8
+        flake8 ci_watson --count
+    # Make sure that packaging will work
+    - name: pep517 build
+      run: |
+        python -m pip install --upgrade setuptools build "twine>=3.3"
+        python -m build --sdist .
+        twine check --strict dist/*
```
Comment (jdavies-st) [inline .github/workflows/ci_workflows.yml:35]: You might want to have it build the wheel too, i.e.

```
python -m build .
```

File: .github/workflows/ci_workflows.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.7'
+    - name: Install and build
```
Comment (jdavies-st) [inline .github/workflows/ci_workflows.yml:48]: Any reason we're running under `python 3.7`?  Maybe we should run under a matrix of python versions?

**Review by jdavies-st — COMMENTED** [review]
Looks good!  Just a few comments below.

Also, I think you can keep around the JenkinsfileRT file so that the RT cron job can run, thus testing the `--bigdata` functionality, which is critical.  I think this PR should just remove/replace Jenkinsfile and the per-PR CI that previously was being handled by ssbjenkins.

TL;DR, don't delete JenkinsfileRT and this PR otherwise looks good.

File: .github/workflows/ci_workflows.yml (hunk 1 above)
Comment (pllim) [reply .github/workflows/ci_workflows.yml:12]: I think I had to define it because somewhere deep in `ci-watson`, it crashes if it is not there.

Comment (pllim) [reply .github/workflows/ci_workflows.yml:12]: But I don't quite remember now, so lemme remove it again and see.

File: .github/workflows/ci_workflows.yml (hunk 3 above)
Comment (pllim) [reply .github/workflows/ci_workflows.yml:48]: I thought Astroconda only works in Python 3.7 and JenkinsfileRT uses Astroconda.

Comment (pllim) [reply .github/workflows/ci_workflows.yml:48]: Between the three jobs, Python 3.7-3.9 are tested here. Making it fancy with matrix/tox can be future work.

**Comment by pllim** [comment]
@jdavies-st , what about now? If it looks good to you, please merge. Thanks!

**Review by jdavies-st — APPROVED** [review]
Looks good!

**Comment by pllim** [comment]
Thanks for the review!
